analyses/copy_number_consensus_call/src/scripts/merged_to_individual_files.py

```diff
@@ -0,0 +1,125 @@
+## Nhat Duong
+## November, 22 2019
+
+######### ASSUMPTIONS ########
+# ../../scratch is available to store intermediate files
+##############################
+
+# Imports in the pep8 order https://www.python.org/dev/peps/pep-0008/#imports
+# Standard library
+import argparse
+import subprocess
+import sys
+import os
+
+# Related third party
+import numpy as np
+import pandas as pd
+
+## Define the extensions
+MANTA_EXT = '.manta'
+CNVKIT_EXT = '.cnvkit'
+FREEC_EXT = '.freec'
+
+# Define the column headers for IDs
+MANTA_ID_HEADER = 'Kids.First.Biospecimen.ID.Tumor'
+CNVKIT_ID_HEADER = 'ID'
+FREEC_ID_HEADER = 'Kids_First_Biospecimen_ID'
+
+parser = argparse.ArgumentParser(description="""This script splits CNV files
+                                                into one per sample. It also
+                                                prints a snakemake config file
+                                                to the specified filename.""")
+parser.add_argument('--manta', required=True,
+                    help='path to the manta file')
+parser.add_argument('--cnvkit', required=True,
+                    help='path to the cnvkit file')
+parser.add_argument('--freec', required=True,
+                    help='path to the freec file')
+parser.add_argument('--snake', required=True,
+                                        help='path new snakemake file')
+parser.add_argument('--maxcnvs', default=2500,
+                    help='samples with more than 2500 cnvs are set to blank')
+parser.add_argument('--cnvsize', default=3000,
+                    help='cnv cutoff size in base pairs')
+parser.add_argument('--freecp', default=0.01,
+                    help='p-value cutoff for freec')
+args = parser.parse_args()
+
+
+## Pandas load/read files in
+merged_manta = pd.read_csv(args.manta, delimiter='\t')
+merged_cnvkit = pd.read_csv(args.cnvkit, delimiter='\t')
+merged_freec = pd.read_csv(args.freec, delimiter='\t')
+
+
+
+## Extract the samples for each files to merge them all together. This takes into account uneven
+## numbers of samples per file
+manta_samples = np.unique(merged_manta[MANTA_ID_HEADER])
+cnvkit_samples = np.unique(merged_cnvkit[CNVKIT_ID_HEADER])
+freec_samples = np.unique(merged_freec[FREEC_ID_HEADER])
+
+
+## Merged and take the unique samples. Any method without a certain sample will get an empty file
+## for of that sample.
+all_samples = np.unique(list(manta_samples) + list(cnvkit_samples)  + list(freec_samples))
+
+## Define and create assumed directories
+manta_d = '../../scratch/manta_manta'
```

Comment:
We can be a bit safer about paths (Doesn't much matter, as we will likely only run this in linux, but for consistency)
```suggestion
manta_d = os.path.join('..', '..', 'scratch', 'manta_manta')
```